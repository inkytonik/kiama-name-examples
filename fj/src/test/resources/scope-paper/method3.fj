// class/method.spt:identity method
class A extends Object {
  A() {
    super();
  }
  B id(B b) {
    return b;
  }
}
class B extends Object {
  B() { super(); }
}
new A().id(new B())
