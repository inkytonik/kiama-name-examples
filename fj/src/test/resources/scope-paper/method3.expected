src/test/resources/scope-paper/method3.fj:6:10:info: b def: B
  B id(B b) {
         ^
src/test/resources/scope-paper/method3.fj:7:12:info: b: B
    return b;
           ^
src/test/resources/scope-paper/method3.fj:13:1:info: new A().id ...: B
new A().id(new B())
^
src/test/resources/scope-paper/method3.fj:13:1:info: new A(): A
new A().id(new B())
^
src/test/resources/scope-paper/method3.fj:13:12:info: new B(): B
new A().id(new B())
           ^
B
